woocommerce/woocommerce-ios
# [Issue Refunds] Create order item refunds

Part of #2843 

# Why

This PR takes care of providing the necessary interfaces to create remote refunds for order items(Shipping refunds will be supported in a next PR).

# How 

- Extracted `RefundableOrderItem` from `IssueRefundViewModel` as it widely used and needed across the refund module.
- Created a `RefundCreationUseCase` type to help transform `OrderItems` into a `Refund` object with `OrderItemRefund` items.
- Added a `localize` method on `CurrencyFormatter` that works with `Decimal` types

# Testing Steps

This code is not yet integrated into the app, but you can try it in the following way.

1. Replace the code in `createRefundConfirmationViewModel` method on `IssueRefundViewController.swift` line 77 with
```swift
func createRefundConfirmationViewModel() -> RefundConfirmationViewModel {

    let items = state.refundQuantityStore.map { RefundableOrderItem(item: $0, quantity: Decimal($1)) }
    let useCase = RefundCreationUseCase(amount: "\(calculateRefundTotal())",
                                        reason: "PR Review",
                                        automaticallyRefundsPayment: false,
                                        items: items,
                                        currencyFormatter: CurrencyFormatter(currencySettings: state.currencySettings))
    let refund = useCase.createRefundObject()

    let action = RefundAction.createRefund(siteID: state.order.siteID,
                                           orderID: state.order.orderID,
                                           refund: refund) { refund, error  in
        print("completed")
        if let error = error {
            print("Error: \(error)")
        }
    }
    ServiceLocator.stores.dispatch(action)

    return RefundConfirmationViewModel(order: state.order, currencySettings: state.currencySettings)
}
```
This will fire a create refund action after tapping the next button in the issue refund screen.

2. Go to an un-refunded order

3. Select some items to refund (Please do not enable refund shipping)

4. Tap Next

5. Go to the sore admin and see that the correct items were refunded for the order

Update release notes:
